src/main/kotlin/com/cognifide/gradle/aem/common/instance/service/pkg/PackageManager.kt

```diff
@@ -298,7 +309,67 @@ class PackageManager(sync: InstanceSync) : InstanceService(sync) {
 
     fun isSnapshot(file: File): Boolean = Patterns.wildcard(file, snapshots.get())
 
-    fun deploy(file: File, activate: Boolean = false) {
+    fun deploy(file: File, activate: Boolean = false): Boolean {
+        if (deployAvoidance.get()) {
+            return deployAvoiding(file, activate)
+        }
+
+        deployRegularly(file, activate)
+        return true
+    }
+
+    private fun deployAvoiding(file: File, activate: Boolean): Boolean {
```

Comment:
Break to shorter functions

Reply:
good working code first; then eventual refactorings.